probe-rs/src/bin/probe-rs/cmd/run.rs

```diff
@@ -92,14 +93,14 @@ impl Cmd {
         let rtt_scan_regions = session.target().rtt_scan_regions.clone();
         let mut core = session.core(0)?;
 
-        if run_download {
-            core.reset_and_halt(Duration::from_millis(100))?;
+        core.reset_and_halt(Duration::from_millis(100))?;
```

Comment:
This changes the behaviour so that we now always reset. We have to keep the old behaviour, and only perform a reset when `run_download` is set. Otherwise the `attach` subcommand will be broken.